// !WITH_NEW_INFERENCE
// !DIAGNOSTICS: -UNUSED_PARAMETER

fun Int.invoke(a: Int) {}
fun Int.invoke(a: Int, b: Int) {}

class SomeClass

fun test(identifier: SomeClass, fn: String.() -> Unit) {
    <!OI;FUNCTION_EXPECTED!><!OI;DEBUG_INFO_MISSING_UNRESOLVED!><!NI;NONE_APPLICABLE!>identifier<!><!><!>()
    <!NI;DEBUG_INFO_MISSING_UNRESOLVED!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!><!NI;DEBUG_INFO_UNRESOLVED_WITH_TARGET!>identifier<!><!>(123)<!>
    <!NI;DEBUG_INFO_MISSING_UNRESOLVED!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!><!NI;DEBUG_INFO_UNRESOLVED_WITH_TARGET!>identifier<!><!>(1, 2)<!>
    1.<!NI;DEBUG_INFO_MISSING_UNRESOLVED!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!><!NI;DEBUG_INFO_UNRESOLVED_WITH_TARGET!>fn<!><!>()<!>
}