// !WITH_NEW_INFERENCE
// !DIAGNOSTICS: -UNUSED_PARAMETER

fun Int.invoke() {}

class SomeClass

fun test(identifier: SomeClass, fn: String.() -> Unit) {
    <!NI;DEBUG_INFO_MISSING_UNRESOLVED!><!OI;FUNCTION_EXPECTED!><!OI;DEBUG_INFO_MISSING_UNRESOLVED!><!NI;UNRESOLVED_REFERENCE_WRONG_RECEIVER!><!NI;DEBUG_INFO_UNRESOLVED_WITH_TARGET!>identifier<!><!><!><!>()<!>
    <!NI;DEBUG_INFO_MISSING_UNRESOLVED!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!><!NI;DEBUG_INFO_UNRESOLVED_WITH_TARGET!>identifier<!><!>(123)<!>
    <!NI;DEBUG_INFO_MISSING_UNRESOLVED!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!><!NI;DEBUG_INFO_UNRESOLVED_WITH_TARGET!>identifier<!><!>(1, 2)<!>
    1.<!NI;DEBUG_INFO_MISSING_UNRESOLVED!><!UNRESOLVED_REFERENCE_WRONG_RECEIVER!><!NI;DEBUG_INFO_UNRESOLVED_WITH_TARGET!>fn<!><!>()<!>
}