// !WITH_NEW_INFERENCE
package f

fun <R> h(<!UNUSED_PARAMETER!>i<!>: Int, <!UNUSED_PARAMETER!>a<!>: Any, <!UNUSED_PARAMETER!>r<!>: R, <!UNUSED_PARAMETER!>f<!>: (Boolean) -> Int) = 1
fun <R> h(<!UNUSED_PARAMETER!>a<!>: Any, <!UNUSED_PARAMETER!>i<!>: Int, <!UNUSED_PARAMETER!>r<!>: R, <!UNUSED_PARAMETER!>f<!>: (Boolean) -> Int) = 1

fun test() = <!OI;CANNOT_COMPLETE_RESOLVE!><!NI;OVERLOAD_RESOLUTION_AMBIGUITY!>h<!><!>(1, 1, 1, { <!OI;CANNOT_INFER_PARAMETER_TYPE!><!OI;UNUSED_ANONYMOUS_PARAMETER!><!NI;UNUSED_ANONYMOUS_PARAMETER!>b<!><!><!> -> 42 })
