// !WITH_NEW_INFERENCE
// !DIAGNOSTICS: -CONFLICTING_JVM_DECLARATIONS
package f

fun <R> h(<!UNUSED_PARAMETER!>f<!>: (Boolean) -> R) = 1
fun <R> h(<!UNUSED_PARAMETER!>f<!>: (String) -> R) = 2

fun test() = <!OI;CANNOT_COMPLETE_RESOLVE!><!NI;OVERLOAD_RESOLUTION_AMBIGUITY!>h<!><!>{ <!OI;CANNOT_INFER_PARAMETER_TYPE!><!OI;UNUSED_ANONYMOUS_PARAMETER!><!NI;UNUSED_ANONYMOUS_PARAMETER!>i<!><!><!> -> getAnswer() }

fun getAnswer() = 42

