// !WITH_NEW_INFERENCE
// See EA-76890 / KT-10843: NPE during analysis
fun lambda(x : Int?) = x?.<!OI;TYPE_INFERENCE_NO_INFORMATION_FOR_PARAMETER!><!OI;FUNCTION_CALL_EXPECTED!><!NO_VALUE_FOR_PARAMETER!><!NI;FUNCTION_CALL_EXPECTED!>let<!><!><!><!> <!OI;DEBUG_INFO_ELEMENT_WITH_ERROR_TYPE!><!NI;UNRESOLVED_REFERENCE!>l<!><!> {
    <!OI;CANNOT_INFER_PARAMETER_TYPE!>y<!> ->
    if (<!OI;DEBUG_INFO_ELEMENT_WITH_ERROR_TYPE!>y<!> <!NI;UNREACHABLE_CODE!><!OI;DEBUG_INFO_ELEMENT_WITH_ERROR_TYPE!><!NI;UNRESOLVED_REFERENCE!>><!><!> 0<!>) <!NI;UNREACHABLE_CODE!>return<!UNRESOLVED_REFERENCE!>@l<!> x<!>
    <!OI;DEBUG_INFO_ELEMENT_WITH_ERROR_TYPE!><!NI;UNREACHABLE_CODE!>y<!><!>
}<!NOT_NULL_ASSERTION_ON_LAMBDA_EXPRESSION!>!!<!>
